airflow/www/security.py

```diff
@@ -16,785 +16,25 @@
 # under the License.
 from __future__ import annotations
 
-import warnings
-from typing import TYPE_CHECKING, Any, Collection, Container, Iterable, Sequence
+from typing import TYPE_CHECKING
 
-from flask import g
-from sqlalchemy import or_, select
-from sqlalchemy.orm import joinedload
+from deprecated import deprecated
 
-from airflow.auth.managers.fab.models import Permission, Resource, Role, User
-from airflow.auth.managers.fab.views.permissions import (
-    ActionModelView,
-    PermissionPairModelView,
-    ResourceModelView,
-)
-from airflow.auth.managers.fab.views.roles_list import CustomRoleModelView
-from airflow.auth.managers.fab.views.user import (
-    CustomUserDBModelView,
-    CustomUserLDAPModelView,
-    CustomUserOAuthModelView,
-    CustomUserOIDModelView,
-    CustomUserRemoteUserModelView,
-)
-from airflow.auth.managers.fab.views.user_edit import (
-    CustomResetMyPasswordView,
-    CustomResetPasswordView,
-    CustomUserInfoEditView,
-)
-from airflow.auth.managers.fab.views.user_stats import CustomUserStatsChartView
-from airflow.exceptions import AirflowException, RemovedInAirflow3Warning
-from airflow.models import DagBag, DagModel
-from airflow.security import permissions
-from airflow.utils.log.logging_mixin import LoggingMixin
-from airflow.utils.session import NEW_SESSION, provide_session
-from airflow.www.extensions.init_auth_manager import get_auth_manager
-from airflow.www.fab_security.sqla.manager import SecurityManager
-from airflow.www.utils import CustomSQLAInterface
-
-EXISTING_ROLES = {
```

Comment:
@vandonr-amz  @vincbeck are we considering removing `EXISTING_ROLES` from here would a breaking change or I'm missing something here?

Reply:
Why would that be a breaking change? In which case user code would use this constant? The only use case I can think of is, if the user provide its own security manager and inside this security manager, they use this constant. But if we want to avoid such things ... every change will be breaking change and doing any change in security managers will be almost impossible.

Or maybe you had something else in mind?

Reply:
> Why would that be a breaking change

yes, it breaking change because I import this constant `from airflow.www.security import EXISTING_ROLES` in my security manager and after this changes my code will break.

> But if we want to avoid such things ... every change will be breaking change and doing

`from airflow.www.security import *` import path was very important for the implementing custom security manager and I feel it would be natural for many people to import `EXISTING_ROLES` if they need this rather than redefine so removing this suddenly is not good and may impact other users. I understand that maintaining backward compatibility for constant type this is painful but still, I would like to keep this maybe we can remove it when we decide to remove  `airflow.www.security.py`

Reply:
Thanks @vincbeck